src/Microsoft.AspNet.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -39,19 +39,14 @@ public override bool CanWriteResult(OutputFormatterContext context, MediaTypeHea
         public override async Task WriteResponseBodyAsync(OutputFormatterContext context)
         {
             var valueAsString = (string)context.Object;
-            if (valueAsString == null)
+            if (string.IsNullOrEmpty(valueAsString))
             {
-                // if the value is null don't write anything.
                 return;
             }
 
             var response = context.ActionContext.HttpContext.Response;
 
-            using (var delegatingStream = new DelegatingStream(response.Body))
-            using (var writer = new StreamWriter(delegatingStream, context.SelectedEncoding, 1024, leaveOpen: true))
-            {
-                await writer.WriteAsync(valueAsString);
-            }
+            await response.WriteAsync(valueAsString, context.SelectedEncoding);
```

Comment:
Verified this is correct. The tricky flush problem was in stream writer, and it is not necessary when writing a string. I *think* the same pattern was added to WriteAsync @Tratcher